css/frontend/src/main/scala/io/udash/css/CssView.scala

```diff
@@ -0,0 +1,80 @@
+package io.udash.css
+
+import io.udash.ReadableProperty
+import io.udash.properties.PropertyCreator
+import org.scalajs.dom
+import org.scalajs.dom.Element
+
+import scalatags.JsDom.all.Modifier
+import scalatags.text.Builder
+
+/** Contains integration of CSS structures with Scalatags. */
+trait CssView {
+  implicit val pcCssStyle: PropertyCreator[CssStyle] = PropertyCreator.propertyCreator[CssStyle]
+  implicit val pcOCssStyle: PropertyCreator[Option[CssStyle]] = PropertyCreator.propertyCreator[Option[CssStyle]]
+  implicit val pcSCssStyle: PropertyCreator[Seq[CssStyle]] = PropertyCreator.propertyCreator[Seq[CssStyle]]
+
+  private object NoopModifier extends Modifier {
+    override def applyTo(t: Element): Unit = ()
+  }
+
+  private class StyleModifier(s: CssStyle) extends Modifier {
+    override def applyTo(t: Element): Unit =
+      s.addTo(t)
+  }
+
+  private class TextStyleModifier(s: CssStyle) extends scalatags.Text.all.Modifier {
+    override def applyTo(t: Builder): Unit =
+      t.appendAttr("class", Builder.GenericAttrValueSource(" " + s.className))
+  }
+
+  implicit class StyleOps(style: CssStyle) {
+    def addTo(element: dom.Element): Unit =
+      element.classList.add(style.className)
+
+    def removeFrom(element: dom.Element): Unit =
+      element.classList.remove(style.className)
+
+    def styleIf(property: ReadableProperty[Boolean]): Modifier =
+      property.reactiveApply(
+        (elem, value) =>
+          if (value) addTo(elem)
+          else removeFrom(elem)
+      )
+
+    def styleIf(condition: Boolean): Modifier =
+      condition match {
```

Comment:
Pattern matching on booleans is not simplified to an if/jump, at least in Scala.